wandb/sdk/internal/internal_api.py

```diff
@@ -3513,6 +3520,63 @@ def replace(a: str, b: str) -> None:
         link_artifact: Dict[str, Any] = response["linkArtifact"]
         return link_artifact
 
+    def _resolve_org_entity_name(self, entity: str, organization: str = "") -> str:
+        # Fetches the org entity of the portfolio entity to
+        # 1. validate the user inputted the correct display org name or org entity name and
+        # 2. return the org entity name so we can use the correct entity name to link the artifact.
+        org_fields = self.server_organization_introspection()
+        can_fetch_org_entity = "orgEntity" in org_fields
+        if not organization and not can_fetch_org_entity:
+            raise ValueError(
+                """Fetching Registry artifacts without inputting an organization is unavailable for your server version.
+                Please upgrade your server to 0.50.0 or later."""
+            )
+        if not can_fetch_org_entity:
+            # Server doesn't support fetching org entity, so we assume the org entity is correctly inputted
+            return organization
+
+        org_entity, org_name = self.fetch_org_entity_from_entity(entity)
+        if organization:
+            if organization != org_name and organization != org_entity:
+                raise ValueError(
+                    f"Artifact belongs to the organization {org_entity!r} and cannot be linked to {organization!r}. Please update the target path with the correct organization name.",
+                )
```

Comment:
Long Python strings can be (and should be!) broken up like this:

```suggestion
                raise ValueError(
                    f"Artifact belongs to the organization {org_entity!r}"
                    f" and cannot be linked to {organization!r}."
                    " Please update the target path"
                    " with the correct organization name.",
                )
```